docs/CMakeLists.txt

```diff
@@ -0,0 +1,105 @@
+    set(extra_search_mappings "")
+  endif()
+
+  # If asked, configure doxygen for the creation of a Qt Compressed Help file.
+  if (LLVM_ENABLE_DOXYGEN_QT_HELP)
+    set(FLANG_DOXYGEN_QCH_FILENAME "org.llvm.flang.qch" CACHE STRING
+      "Filename of the Qt Compressed help file")
+    set(FLANG_DOXYGEN_QHP_NAMESPACE "org.llvm.flang" CACHE STRING
+      "Namespace under which the intermediate Qt Help Project file lives")
+    set(FLANG_DOXYGEN_QHP_CUST_FILTER_NAME "FLANG ${FLANG_VERSION}" CACHE STRING
+      "See http://qt-project.org/doc/qt-4.8/qthelpproject.html#custom-filters")
+    set(FLANG_DOXYGEN_QHP_CUST_FILTER_ATTRS "FLANG,${FLANG_VERSION}" CACHE STRING
+      "See http://qt-project.org/doc/qt-4.8/qthelpproject.html#filter-attributes")
+    set(flang_doxygen_generate_qhp "YES")
+    set(flang_doxygen_qch_filename "${FLANG_DOXYGEN_QCH_FILENAME}")
+    set(flang_doxygen_qhp_namespace "${FLANG_DOXYGEN_QHP_NAMESPACE}")
+    set(flang_doxygen_qhelpgenerator_path "${LLVM_DOXYGEN_QHELPGENERATOR_PATH}")
+    set(flang_doxygen_qhp_cust_filter_name "${FLANG_DOXYGEN_QHP_CUST_FILTER_NAME}")
+    set(flang_doxygen_qhp_cust_filter_attrs "${FLANG_DOXYGEN_QHP_CUST_FILTER_ATTRS}")
+  else()
+    set(flang_doxygen_generate_qhp "NO")
+    set(flang_doxygen_qch_filename "")
+    set(flang_doxygen_qhp_namespace "")
+    set(flang_doxygen_qhelpgenerator_path "")
+    set(flang_doxygen_qhp_cust_filter_name "")
+    set(flang_doxygen_qhp_cust_filter_attrs "")
+  endif()
+
+  option(LLVM_DOXYGEN_SVG
+    "Use svg instead of png files for doxygen graphs." OFF)
+  if (LLVM_DOXYGEN_SVG)
+    set(DOT_IMAGE_FORMAT "svg")
+  else()
+    set(DOT_IMAGE_FORMAT "png")
+  endif()
+
+  configure_file(${CMAKE_CURRENT_SOURCE_DIR}/doxygen.cfg.in
+    ${CMAKE_CURRENT_BINARY_DIR}/doxygen.cfg @ONLY)
+
+  set(abs_top_srcdir)
+  set(abs_top_builddir)
+  set(DOT)
+  set(enable_searchengine)
+  set(searchengine_url)
+  set(enable_server_based_search)
+  set(enable_external_search)
+  set(extra_search_mappings)
+  set(flang_doxygen_generate_qhp)
+  set(flang_doxygen_qch_filename)
+  set(flang_doxygen_qhp_namespace)
+  set(flang_doxygen_qhelpgenerator_path)
+  set(flang_doxygen_qhp_cust_filter_name)
+  set(flang_doxygen_qhp_cust_filter_attrs)
+  set(DOT_IMAGE_FORMAT)
+
+  add_custom_target(doxygen-flang
+    COMMAND ${DOXYGEN_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/doxygen.cfg
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+    COMMENT "Generating flang doxygen documentation." VERBATIM)
+
+#  if (LLVM_BUILD_DOCS)
+#   add_dependencies(doxygen doxygen-flang)
+#  endif()
+
+  if (NOT LLVM_INSTALL_TOOLCHAIN_ONLY)
+    install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/doxygen/html
+      DESTINATION docs/html)
+  endif()
+endif()
+endif()
+
+if (LLVM_ENABLE_SPHINX)
+  include(AddSphinxTarget)
+  if (SPHINX_FOUND)
+    if (${SPHINX_OUTPUT_HTML})
+      add_sphinx_target(html flang)
+    endif()
+    if (${SPHINX_OUTPUT_MAN})
+      add_sphinx_target(man flang)
+    endif()
+  endif()
```

Comment:
I'm not sure if this Sphinx stuff will work yet as we don't have any Sphinx docs

Reply:
Ok, should I comment it as of now if we are not currently interested in Sphinx?

Reply:
The Sphinx stuff worked to generate the doxygen html files.

Reply:
Oh yes I see, this should be left in then

Reply:
Ok, @sscalpone what target did you use for the sphinx stuff?
Was it `ninja doxygen-flang`, and instead of Sphinx related docs you got doxygen docs as we don't have any Sphinx docs, is it?

Reply:
@sammeranjoshi How are the API documents generated?

Reply:
The API docs will be generated by doxygen and not touch Sphinx, sphinx is just for RST documentation; e.g. https://llvm.org/docs/ this page is generated by sphinx, but https://llvm.org/doxygen/ this page is generated by doxygen. They aren't connected to each other (other than by hyperlinks on the pages).